File: 2022.01.20/python/11729.py
n = int(input());
def hanoi(n,start,end,way_point):
  if n == 1:
    print(start,end)
    return
  hanoi(n-1,start,way_point,end)
  print(start,end)
  hanoi(n-1,way_point,end,start)
print(2**n-1)
hanoi(n,1,3,2)

File: Boj.단계별/10_재귀/javascript/11729.js
const n = Number(require('fs').readFileSync('/dev/stdin').toString().split('\n')[0]);
let result;
let hanoi = (n,start,end,wayPoint,result) => {
  if (n===0) {
    return result;
  }
  result += hanoi(n-1,start,wayPoint,end,"");
  result += `${start} ${end}\n`;
  result += hanoi(n-1,wayPoint,end,start,"");
  return result;
}
console.log(2**n-1);
result = hanoi(n,1,3,2,"");
console.log(result);